lib/location.js

```diff
@@ -166,13 +170,22 @@ var LocationState = State.extend(function (self, name, opts) {
             });
     };
 
+
     // Handler for user response to refine location
     self.handlers.refinequestion = function(content) {
-        var handler = {
+        var choices = {
             n: self.handlers.refinequestion.next,
             p: self.handlers.refinequestion.prev
-        }[content];
+        };
+
+        if (self.skip_text){
+            choices.s = self.handlers.skip;
+        }
+        if (self.retry_text){
```

Comment:
Very nitpicky thing -- we don't have a space between `){` in these two, but we do elsewhere.